### PR TITLE
feat: New Robustness Regression Attacker: Ridge Poissoner

### PR DESCRIPTION
**Pull request by fracarfer5**
Adding ridge regression-based poisoner, updating tutorial and API references. Furthermore, two initializers were added.

### CODE REVIEW

File: src/holisticai/robustness/attackers/regression/gb_base.py [hunk 1]
```diff
@@ -41,7 +44,19 @@ class GDPoisoner:
     .. [1] Jagielski, M., Oprea, A., Biggio, B., Liu, C., Nita-Rotaru, C., & Li, B. (2018, May). Manipulating machine learning: Poisoning attacks and countermeasures for regression learning. In 2018 IEEE symposium on security and privacy (SP) (pp. 19-35). IEEE.
```
Comment (crismunoz) [inline src/holisticai/robustness/attackers/regression/gb_base.py:47]: add description for initializer

Comment (crismunoz) [reply src/holisticai/robustness/attackers/regression/gb_base.py:47]: check all child classes too

**Review by crismunoz — CHANGES_REQUESTED** [review]

**Review by crismunoz — APPROVED** [review]

File: src/holisticai/robustness/attackers/regression/gb_attackers.py [hunk 2]
```diff
@@ -270,3 +276,282 @@ def _comp_attack_trn(self, clf, wxc, bxc, wyc, byc, otherargs):  # noqa: ARG002
...
+    Parameters
+    ----------
+    poison_proportion : float
+        The proportion of points to flip. Default is 0.2.
+    num_inits : int
+        The number of initializations. Default is 1.
+    max_iter : int
+        The maximum number of iterations. Default is 15.
+    initializer : str
+        The initialization method. Default is 'inf_flip'.
```
Comment (crismunoz) [inline src/holisticai/robustness/attackers/regression/gb_attackers.py]: tem que espeficicar que parametros podem ser de entrada

**Review by crismunoz — CHANGES_REQUESTED** [review]

File: src/holisticai/robustness/attackers/regression/gb_attackers.py (hunk 2 above)
Comment (fracarfer5) [reply src/holisticai/robustness/attackers/regression/gb_attackers.py]: Done. Added possible options for initializers

**Review by crismunoz — APPROVED** [review]

**Pull request merged**